system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php

```diff
@@ -389,16 +389,19 @@ protected function setupEntries()
      */
     private function createAuthorFilter($channel_id = null)
     {
-        $db = ee('db')->distinct()
-            ->select('t.author_id, m.screen_name, m.username')
-            ->from('channel_titles t')
-            ->join('members m', 'm.member_id = t.author_id', 'LEFT')
-            ->order_by('screen_name', 'asc');
+        // It is more performant to query the members table and filter by having at least one channel_title authored
+        // than to do a distinct query on author_id across a potentially very large channel_titles table.
+        $where = 'SELECT count(t.entry_id) FROM '. ee()->db->dbprefix('channel_titles').' t WHERE t.author_id = m.member_id';
 
         if ($channel_id) {
-            $db->where('channel_id', $channel_id->channel_id);
+            $where .= ' AND t.channel_id = '. (int) $channel_id->channel_id;
```

Comment:
Looks like $channel_id is not actually a channel_id. I'd assume it's an instance of the channel model based on this code? While we're editing this, it might be worth updating the name of the variable to better reflect this.